### PR TITLE
[Logstage] Add raw namespace to loggers

### PR DESCRIPTION
**Pull request by catostrophe**
This is the first part. Will do the same for IO loggers after review.

### CODE REVIEW

**Comment by catostrophe** [comment]
Resolves #1164

File: logstage/logstage-core/src/main/scala/izumi/logstage/api/strict/IzStrictLogger.scala [hunk 1]
```diff
@@ -3,22 +3,25 @@ package izumi.logstage.api.strict
...
   override val customContext: Log.CustomContext,
 ) extends RoutingLogger
   with AbstractMacroStrictLogger {
 
-  override def withCustomContext(context: CustomContext): IzLogger = new IzLogger(router, customContext + context)
-  final def withCustomContext(context: (String, StrictEncoded)*): IzLogger = withCustomContextMap(context.toMap)
-  final def withCustomContextMap(context: Map[String, StrictEncoded]): IzLogger = withCustomContext(CustomContext.fromMap(context))
+  override type Self = IzStrictLogger
 
-  final def apply(context: CustomContext): IzLogger = withCustomContext(context)
```
Comment (catostrophe) [inline logstage/logstage-core/src/main/scala/izumi/logstage/api/strict/IzStrictLogger.scala:18]: Notice an issue here: a strict logger could become unstrict. That's why (and not only) I introduced the `Self` type.

**Review by pshirshov — APPROVED** [review]

**Review by neko-kai — APPROVED** [review]
Thank you! 👍

File: logstage/logstage-core/src/main/scala/izumi/logstage/macros/LoggerMacroMethods.scala [hunk 2]
```diff
@@ -1,76 +1,107 @@
...
+    }
+
     c.universe.reify {
-      {
-        val self = c.prefix.splice
-        if (self.acceptable(Log.LoggerId(CodePositionMaterializerMacro.getApplicationPointId(c).splice), level.splice)) {
-          self.unsafeLog(Log.Entry.create(level.splice, message.splice)(position.splice))
-        }
-      }
+      c.prefix.splice.log(l.splice)(m.splice)(getEnclosingPosition(c).splice)
```
Comment (neko-kai) [inline logstage/logstage-core/src/main/scala/izumi/logstage/macros/LoggerMacroMethods.scala:104]: @catostrophe Ergh, just noticed this, this change degrades performance (inline if is faster than a call)

Comment (catostrophe) [reply logstage/logstage-core/src/main/scala/izumi/logstage/macros/LoggerMacroMethods.scala:104]: Ouch. Sorry about that. Can't even remember why I touched it...

Comment (neko-kai) [reply logstage/logstage-core/src/main/scala/izumi/logstage/macros/LoggerMacroMethods.scala:104]: No worries

**Pull request merged**